sd3_infer.py

```diff
@@ -294,21 +382,29 @@ def do_sampling(
         self.print("Sampling done")
         return latent
 
-    def vae_encode(self, image) -> torch.Tensor:
+    def vae_encode(self, image, using_8b_controlnet: bool = False) -> torch.Tensor:
         self.print("Encoding image to latent...")
         image = image.convert("RGB")
         image_np = np.array(image).astype(np.float32) / 255.0
         image_np = np.moveaxis(image_np, 2, 0)
         batch_images = np.expand_dims(image_np, axis=0).repeat(1, axis=0)
-        image_torch = torch.from_numpy(batch_images)
-        image_torch = 2.0 * image_torch - 1.0
+        image_torch = torch.from_numpy(batch_images).cuda()
+        if using_8b_controlnet:
+            image_torch = 2.0 * image_torch - 1.0
```

Comment:
I think this part is wrong?
This line is supposed to be within the non-controlnet case (as well?), right?